codegen/cmd/injection-gen/generators/reconciler_controller.go

```diff
@@ -164,7 +164,7 @@ const (
 // the queue through an implementation of {{.controllerReconciler|raw}}, delegating to
 // the provided Interface and optional Finalizer methods. OptionsFn is used to return
 // {{.controllerOptions|raw}} to be used but the internal reconciler.
-func NewImpl(ctx {{.contextContext|raw}}, r Interface{{if .hasClass}}, classValue string{{end}}, optionsFns ...{{.controllerOptionsFn|raw}}) *{{.controllerImpl|raw}} {
```

Comment:
We had looked at this before and it is too easy for multiple controllers to do this wrong and both race to reconcile a resource.

/hold